pkg/controller/mongodb/mongodb_controller.go

```diff
@@ -79,6 +81,22 @@ func (r *ReplicaSetReconciler) Reconcile(request reconcile.Request) (reconcile.R
 		return reconcile.Result{}, err
 	}
 
+	// TODO: Read current automation config version from config map
```

Comment:
I would not even care about doing this in the controller. This automation config implementation is completely independent of kubernetes and as such we should be able to get to a comfortable level with it only by unit testing it.

In fact the automationconfig Builder should return a struct that we will pass to the next "stage". That "next stage" will be in charge of creating the ConfigMap, as we have discussed... but that's should be left for a future patch